src/DssSpell.t.base.sol

```diff
@@ -266,22 +266,53 @@ contract DssSpellTestBase is Config, DssTest {
     function setUp() public {
         setValues(address(chief));
 
-        spellValues.deployed_spell_created = spellValues.deployed_spell != address(0) ? spellValues.deployed_spell_created : block.timestamp;
+        if (spellValues.deployed_spell != address(0)) {
+            spell = DssSpell(spellValues.deployed_spell);
+            if (spell.eta() != 0) {
+                // if we have a scheduled spell in the config
+                // we want to roll our fork to the block where it was deployed
+                // this means the test suite will continue to accurately pass/fail
+                // even if mainnet has already scheduled/cast the spell
+                vm.makePersistent(address(this));
+                vm.makePersistent(address(rates));
+                vm.makePersistent(address(addr));
+                vm.makePersistent(address(deployers));
+                _rollToDeployedSpellBlock(address(spell));
+            }
+        } else {
+            spell = new DssSpell();
+        }
+
         _castPreviousSpell();
-        spell = spellValues.deployed_spell != address(0) ?
-            DssSpell(spellValues.deployed_spell) : new DssSpell();
-
-        if (spellValues.deployed_spell_block != 0 && spell.eta() != 0) {
-            // if we have a deployed spell in the config
-            // we want to roll our fork to the block where it was deployed
-            // this means the test suite will continue to accurately pass/fail
-            // even if mainnet has already scheduled/cast the spell
-            vm.makePersistent(address(this));
-            vm.makePersistent(address(rates));
-            vm.makePersistent(address(addr));
-            vm.makePersistent(address(deployers));
-            vm.rollFork(spellValues.deployed_spell_block);
+    }
+
+    function _rollToDeployedSpellBlock(address _spell) internal {
+        if (_spell == address(0)) {
+            vm.rollFork(block.number);
+            return;
+        }
+
+        uint256 low = 5273074; // MCD_VAT Deployed Aug-06-2021 04:26:38 PM +UTC
+        uint256 high = block.number;
+        uint256 mid;
+
+        while (low != high) {
+            mid = (low + high) / 2;
+            vm.rollFork(mid);
+            if (_isContractDeployed(_spell)) {
+                high = mid - 1;
+            } else {
+                low = mid + 1;
+            }
         }
+
+        vm.rollFork(low + 1);
```

Comment:
Tested:
- `make test` PASS
- block [8451434](https://goerli.etherscan.io/block/8451434) deployed PASS
- block [8451989](https://goerli.etherscan.io/block/8451989) cast PASS
- block [8451982](https://goerli.etherscan.io/block/8451982) schedule, gets stuck

Then, playing around I found this pattern:

- 8451982, gets stuck `make test block=8451982`
- 8451983 pass
- 8451984 fail
- 8451985 stuck
- 8451986 pass
- 8451987 fail
- 8451988 stuck
- ...

Reply:
Addressed in #167

Reply:
Thank you!